atlasdb-impl-shared/build.gradle

```diff
@@ -17,6 +17,7 @@ dependencies {
   compile project(":timestamp-api")
   compile project(":timestamp-client")
 
+  compile 'com.github.ben-manes.caffeine:caffeine'
```

Comment:
nit: can we do `implementation`? `compile` is the legacy thing

Reply:
Will push a follow up PR to cleanup a bunch of the gradle dependencies